src/geom/prep/PreparedPolygonPredicate.cpp

```diff
@@ -37,76 +38,79 @@ namespace prep { // geos.geom.prep
 //
 // protected:
 //
+struct AnyMatchingLocationFilter : public CoordinateFilter {
```

Comment:
Isn't this doing a lot more work than the original use of `ComponentCoordinateExtracter`?  This will test **every** coordinate in the geometry, not just a single point from each component.

Reply:
Yes. Fixed.

Reply:
And then really need an `GeometryComponentFilter.isDone()` short-circuit.  

Or at least a check for the short-circuit condition in the `filter()` method to avoid calling `locate()` if done.

Reply:
The short-circuit check is there, but would be better moved to `isDone`. Will do this.

Reply:
Hmm @dr-jts , maybe I don't understand `GeometryComponentFilter`. Why does a `GeometryCollection` apply the filter to itself? https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/geom/GeometryCollection.java#L205

Reply:
Actually _every_ Geometry subclass applies `GeometryComponentFilter` to itself first.  

Which is contrary to the `GeometryComponentFilter` doc.  Not sure why this is...  I suspect the idea was that it was easy to filter out complex types if needed.  But maybe this is confusing and should be eliminated.  

In that case I guess this change should use `GeometryFilter` to extract a candidate coordinate from each connected component.  But need to verify that this works correctly and doesn't miss anything.

This is muddying the waters a bit.. sorry bout that.